internal/unlink/unlink.go

```diff
@@ -0,0 +1,216 @@
+package unlink
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/BurntSushi/toml"
+	"github.com/go-errors/errors"
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+	"github.com/supabase/cli/internal/migration/repair"
+	"github.com/supabase/cli/internal/utils"
+	"github.com/supabase/cli/internal/utils/credentials"
+	"github.com/supabase/cli/internal/utils/tenant"
+	"github.com/supabase/cli/pkg/api"
+)
+
+var updatedConfig ConfigCopy
+
+type ConfigCopy struct {
+	Api    interface{} `toml:"api"`
+	Db     interface{} `toml:"db"`
+	Pooler interface{} `toml:"db.pooler"`
+}
+
+func (c ConfigCopy) IsEmpty() bool {
+	return c.Api == nil && c.Db == nil && c.Pooler == nil
+}
+
+func PreRun(projectRef string, fsys afero.Fs) error {
+	return utils.LoadConfigFS(fsys)
+}
+
+func Run(ctx context.Context, password string, fsys afero.Fs, options ...func(*pgx.ConnConfig)) error {
+	unlinkServices(ctx, fsys)
+
+	// 3. Save project ref
+	return utils.WriteFile(utils.ProjectRefPath, []byte(""), fsys)
+}
+
+func PostRun(projectRef string, stdout io.Writer, fsys afero.Fs) error {
+	fmt.Fprintln(stdout, "Finished "+utils.Aqua("supabase unlink")+".")
+	if updatedConfig.IsEmpty() {
+		return nil
+	}
+	fmt.Fprintln(os.Stderr, "Local config differs from linked project. Try updating", utils.Bold(utils.ConfigPath))
+	enc := toml.NewEncoder(stdout)
+	enc.Indent = ""
+	if err := enc.Encode(updatedConfig); err != nil {
+		return errors.Errorf("failed to marshal toml config: %w", err)
+	}
+	return nil
+}
+
+func unlinkServices(ctx context.Context, fsys afero.Fs) {
```

Comment:
Since unlink is a purely local operation, we can simply delete `utils.TempDir` without making any api requests.

Reply:
@sweatybridge i am working on a fix for this. The idea is to remove the `.temp` dir and input the `project-ref` to be able to remove the credentials for the project.

Reply:
@sweatybridge could you have a look at the [tests](https://github.com/supabase/cli/pull/1849/files#diff-f021fb6dc36f857212d1904d19a99399e15e565d4116eefa2971f66dca9f5c8fR13) i have written. they seem to fail due to:

```
        	Test:       	TestUnlinkCommand/unlink_valid_project
Removing credentials for project test-project...
    unlink_test.go:33: 
        	Error Trace:	/home/runner/work/cli/cli/internal/unlink/unlink_test.go:33
        	Error:      	Received unexpected error:
        	            	failed to remove credentials for project 'test-project': failed to set credentials: The name org.freedesktop.secrets was not provided by any .service files
```

Reply:
I think you need to call `keyring.MockInit()` in this test suite. For eg.

https://github.com/supabase/cli/blob/main/internal/logout/logout_test.go#L19